### PR TITLE
Implemented new anisotropic material::properties and the allowed for reading

### LINKED ISSUE #252: Add conditional to read new material properties read_materials
TODO

- [x] Add elastic anisotropic property to     
  ```cpp
  specfem::material::properties<specfem::element::medium_tag::elastic,
                                specfem::element::property_tag::isotropic>
  ```
- [x] Add conditional to `read_materials` to read the anisotropic material.

### PR DESCRIPTION
**Pull request by lsawade**
# Description

See linked issue

# Issue Number

closes #252 

# Checklist

Please make sure to check developer documentation on specfem docs. 

- [x] I ran the code through pre-commit to check style
- [ ] My code passes all the integration tests
- [x] I have added sufficient unittests to test my changes
- [x] I have added/updated documentation for the changes I am proposing
- [x] I have updated CMakeLists to ensure my code builds
- [x] My code builds across all platforms

### CODE REVIEW

**Comment by lsawade** [comment]
closes #252

**Review by icui — COMMENTED** [review]
These are parameters for 2D only, should we rename everything to anisotropic2d?

**Comment by lsawade** [comment]
Properties etc should probably rather be of dimension type

**Comment by icui** [comment]
> Properties etc should probably rather be of dimension type. Ie be templated

I think material property should be dimension-independent. This 2D parameterization can be used for 3D simulation as well and vice versa (2d simulation taking 21-parameter model should behave the same as the 6-parameter model).

File: include/material/elastic_properties.hpp [hunk 1]
```diff
@@ -110,5 +110,109 @@ class properties<specfem::element::medium_tag::elastic,
   type_real poisson;         ///< Poisson's ratio
 };
 
+/**
+ * @brief Template specialization for elastic anisotropic material properties
+ *
+ */
+template <>
```
[the remark this answers is not in the document]
Comment (Rohit-Kakodkar) [reply include/material/elastic_properties.hpp:117]: Pushed this to later refactor

**Review by Rohit-Kakodkar — APPROVED** [review]

**Comment by lsawade** [comment]
Pushed the namespace removal.

**Pull request merged**